clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1211,12 +1211,9 @@ protected void updateOffsets(
       // DaVinci clients don't need to maintain leader production states
       if (!isDaVinciClient) {
         // also update the leader topic offset using the upstream offset in ProducerMetadata
-        if (kafkaValue.producerMetadata.upstreamOffset >= 0
-            || (kafkaValue.leaderMetadataFooter != null && kafkaValue.leaderMetadataFooter.upstreamOffset >= 0)) {
+        if (kafkaValue.leaderMetadataFooter != null && kafkaValue.leaderMetadataFooter.upstreamOffset >= 0) {
```

Comment:
Is this change desired?
I am wondering whether this may or may not cause any problem if there are some legacy version topics, which are still using the legacy protocols and in the rebootstraping/cluster recovery scenario, the offset checkpointing logic won't work.

cc @huangminchn to double confirm it.

Reply:
We discussed, decide to give up bundling `producerMetadata` cleanup inside this KME upgrade.

Reply:
Yeah, there could be legacy version topics, unless we do some auditing.